core/src/modules/ops/destination.ts

```diff
@@ -912,13 +913,41 @@ export namespace DestinationOps {
     success: boolean;
     error: Error;
   }> {
-    const _exports: Export[] = []; // only ones we are sending
+    const _exports: Export[] = [];
+
     for (const _export of givenExports) {
-      // TODO: maybe should recalcuate hasChanges and from current mostRecent
-      if (!_export.hasChanges) {
-        await _export.complete(); // do not include exports with hasChanges=false
+      const { isLocked, lockedBy } = await getLock(
+        `${_export.recordId}:${_export.destinationId}`,
+        _export.id,
+        null,
+        "export"
+      );
+
+      if (isLocked) {
+        const lockingExport = await Export.findById(lockedBy);
+        //if the export we're looking at is older than the one that locked this pair, cancel it
+        if (lockingExport.createdAt > _export.createdAt) {
+          await _export.update({
+            state: "canceled",
+            sendAt: null,
+            errorMessage: `Outdated duplicate of export ${lockingExport.id}`,
+            errorLevel: null,
+            completedAt: new Date(),
+          });
+          return;
+        } else if (lockingExport.id !== _export.id) {
+          //if the export we're looking at didn't create this lock, keep it pending for now
+          return;
```

Comment:
This return would exit the whole function. perhaps `continue` instead to go to the next _export in the loop